game.py

```diff
@@ -12,16 +12,40 @@
     '-----------'
 ]
 
-
 def snowman(snowman_word):
     """Complete the snowman function
     replace "pass" below with your own code
     It should print 'Congratulations, you win!'
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    wrong_guesses_count = 0
+    wrong_guesses_list = []
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    
+
+    while wrong_guesses_count < SNOWMAN_MAX_WRONG_GUESSES:
+        print_snowman_graphic(wrong_guesses_count)
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+        user_guess = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
 
+        if user_guess in correct_letter_guess_statuses:
+            print ("correct guess")
+            correct_letter_guess_statuses[user_guess] = True
+    
+        else:
+            print("wrong guess")
+            wrong_guesses_list.append(user_guess)
+            wrong_guesses_count += 1
+
+        if is_word_guessed(snowman_word, correct_letter_guess_statuses):
+            print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+            print("Congratulations, you win!")
+            return
+        
+    print_snowman_graphic(wrong_guesses_count)
+    print(f"Sorry, you lose! The word was {snowman_word}")
+    return
```

Comment:
We don't need an explicit `return` here. When the end of the function is reached, the function automatically returns. We should use a return statement
1. if we need to return an actual value, or
2. if we are returning before the end of the function

The return _was_ needed when this code was located within the loop, but with it moved out here, we can remove it.